internal/services/containers/container_group_resource.go

```diff
@@ -72,14 +72,6 @@ func resourceContainerGroup() *pluginsdk.Resource {
 				}, false),
 			},
 
-			"network_profile_id": {
```

Comment:
sadly we cannot just remove this property, if its still in the API we need to deprecate it and then wrap its removal in a 4.0 beta flag

Reply:
Do you mean something like this - https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/bot/bot_channels_registration_resource.go#L171 ?

That field is replaced with subnet in newer API so it would not be possible to use it once we migrate to hat API.
Or we should just set `Deprecated: "..."` to mark `network_profile_id` as deprecated field?

Reply:
@katbyte @magodo What should be done to be able to merge this one? Can you, please, give more details? Thanks

Reply:
yep @dkuzmenok - just like that, as its removed from the API we can just remove all code around it and direct users to use the new property via the deprecation message.